lib/LaTeXML/Core/Mouth.pm

```diff
@@ -288,49 +284,53 @@ sub readToken {
       $$self{lineno}++;
       $$self{colno} = 0;
       my $line = $self->getNextLine;
+      # For \read, we have to return something for EOL, and handle implicit final newline
+      my $read_mode = (($STATE->lookupValue('PRESERVE_NEWLINES') || 0) > 1);
+      my $eolch     = "\r";
+      if (my $eol = $STATE->lookupDefinition(T_CS('\endlinechar'))) {
+        $eol   = $eol->valueOf()->valueOf;
+        $eolch = (($eol > 0) && ($eol <= 255) ? chr($eol) : undef); }
```

Comment:
Possibly, but would still require a careful testing to avoid a Perl trap. I used `\enlinechar` being "0" to force myself to get it (partly) right, since "" and "0" are both "false" from Perl's perspective.